genrl/marl/maa2c/a2c_agent.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import torch 
+import torch.nn as nn
+import torch.optim as optim
+import torch.autograd as autograd
+from torch.autograd import Variable
+from torch.distributions import Categorical
+from a2c import *
+import torch.nn.functional as F
+import gc
+
+class A2CAgent:
+
+  def __init__(self,env,lr=2e-4,gamma=0.99):
+    self.env = env
+    self.lr = lr
+    self.gamma = gamma
+
+    self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    
+    self.num_agents = self.env.n
+
+    self.input_dim = env.observation_space[0].shape[0]
+    self.action_dim = self.env.action_space[0].n
+    self.actorcritic = CentralizedActorCritic(self.input_dim,self.action_dim).to(self.device)
+    model_path = ".tests/models/four_agents/actorcritic_network_no_comms_discounted_rewards_lr_2e-4_with_grad_norm_0.5_entropy_pen_0.008_xavier_uniform_init_clamp_logs.pt"
+    self.actorcritic.load_state_dict(torch.load(model_path,map_location=torch.device('cpu')))
+
+    self.MSELoss = nn.MSELoss()
+    self.actorcritic_optimizer = optim.Adam(self.actorcritic.parameters(),lr=lr)
+
+  def get_action(self,state):
+    state = torch.FloatTensor(state).to(self.device)
+    logits,_ = self.actorcritic.forward(state)
+    # print("logits:",logits)
+    del state
+    dist = F.softmax(logits,dim=0)
+    # print("dist:",dist)
+    del logits
+    # print('dist: ', dist)
+    probs = Categorical(dist)
+    # print("PROBS:",probs)
+    index = probs.sample().cpu().detach().item()
+    
+    return index
```

Comment:
This can be shifted to 'centralised_actor_critic` class if `get_action` will look similar in all algorithms.

Reply:
It is environment dependent but yeah it can done.